main.py

```diff
@@ -0,0 +1,34 @@
+import streamlit as st
+from errors import FailedToDownloadException, EmptyStringException, InvalidURL
+from utils import get_and_save_image, get_class
+import cv2
+
+
+st.title("Image classifier")
+image_url = st.text_input("Image URL")
+error = ""
+
+if st.button("Classify!", key="enter"):
+    try:
```

Comment:
en plus simple, je pense que tu aurais utiliser ça pour lire l'image directement avec PIL
```
from PIL import Image
import requests

im = Image.open(requests.get(url, stream=True).raw)
```
cf  https://stackoverflow.com/questions/7391945/how-do-i-read-image-data-from-a-url-in-python

Reply:
(l'autre avantage est que ça ne stock pas sur ton dd l'image, ce qui pour une appli sur ton ordi n'est pas impactant mais qui pourrait gêner si on faisait ce genre d'apps dans des "lambdas")